In [1]:
import numpy as np
import matplotlib.pyplot as plt
import yfinance as yf
import pandas as pd

# TAREA 3

Asume que eres el Portfolio Manager de Banamex y deseas comparar el desempeño de una estrategia de inversión activa con una pasiva (Benchmark), realizando un análisis básico del riesgo y rendimiento de ambas estrategias. Sigue los pasos para evaluarlas:

---

## 1. Estrategia de inversión activa
Tu estrategia de inversión activa contiene las siguientes acciones con ponderaciones equitativas para tu portafolio:

- Berkshire Hathaway B  
- Visa  
- Goldman Sachs Group Inc  
- Progressive Corp  

### Actividades:
- Investiga los tickers de cada acción, su sector y modelo de negocio, dando un resumen ejecutivo de estas. (10%)  
- Descarga los precios de cierre ajustado desde el **2018-01-01** hasta el **2025-06-03**.  
- Calcula las métricas de riesgo y rendimientos anuales, ratio de Sharpe y la matriz de correlación. (20%)  
- Interpreta la matriz de correlación. (10%)  

## **Berkshire Hathaway Inc. (BRK-B)**

**Actividad y sector:**

Berkshire Hathaway Inc. es un holding conglomerado que posee decenas de subsidiarias diversificadas. Sus actividades abarcan principalmente seguros y reaseguros, transporte ferroviario de carga, servicios públicos de energía (electricidad y gas) y una amplia gama de negocios de manufactura, servicios y comercio minorista. En seguros, por ejemplo, controla compañías reconocidas como GEICO (automóviles) y General Re (reaseguro). En transporte, opera BNSF Railway, uno de los mayores ferrocarriles de Norteamérica. También es dueña de Berkshire Hathaway Energy, un consorcio de utilidades eléctricas, gasíferas y proyectos de energías renovables, entre otras empresas industriales. Por esta razón, incluir BRK-B en un portafolio ofrece amplia diversificación automática, pues abarca tantos sectores que a menudo se la compara con un fondo de inversión en sí mismo

**Fuentes de Ingresos**

- Seguros (Insurance): Es la mayor fuente de ingresos de Berkshire. Incluye primas de seguros de propiedad y accidentes y resultados de inversión del “float” de seguros. En 2023 este segmento generó aproximadamente US$95,022 millones en ingresos.
- Ferrocarriles (Burlington Northern Santa Fe, BNSF): Opera una de las principales redes ferroviarias de carga en EE.UU. Los ingresos de BNSF rondaron los US$23,876 millones en 2023, provenientes del transporte de mercancías (granos, productos industriales, bienes de consumo, etc.).
- Energía (Berkshire Hathaway Energy): Incluye compañías de electricidad (ej. PacifiCorp, MidAmerican) y gas natural, así como parques eólicos/solares. Generó cerca de US$26,008 millones en ingresos en 2023. Sus utilidades operativas pueden variar según regulaciones y precios de la energía.

**Fuentes:**

- [Berkshire Hathaway Inc: Overview](https://www.globaldata.com/company-profile/berkshire-hathaway-inc/#:~:text=Berkshire%20Hathaway%20Inc%20(Berkshire)%20is,Omaha%2C%20Nebraska%2C%20the%20US.)
- [Business segment data](https://www.sec.gov/Archives/edgar/data/1067983/000095017024019719/R63.htm#:~:text=BHE)

## **Visa Inc. (V)**

**Actividad y sector:**

Visa Inc. es una compañía global de tecnología de pagos que conecta a consumidores, empresas, instituciones financieras y gobiernos en más de 200 países, facilitando pagos electrónicos rápidos, seguros y confiables. Opera VisaNet, una de las redes de procesamiento de transacciones más avanzadas del mundo (capaz de manejar ~65 mil mensajes de transacción por segundo) con altos estándares de seguridad anti-fraude. Cabe destacar que Visa no es un banco: no emite tarjetas ni otorga crédito directo a los consumidores, sino que provee la infraestructura y servicios de procesamiento (autorización, compensación y liquidación de transacciones) para sus instituciones financieras afiliadas y comercios. Su modelo de negocio se basa en cobrar tarifas por el uso de esta red y servicios asociados, en lugar de intereses o préstamos y sus principales competidores son otras redes de pago como Mastercard y American Express, así como empresas de pagos digitales (por ejemplo, PayPal).

**Fuentes de Ingresos:**

En total, Visa reportó USD 32.7 mil millones en ingresos netos durante 2023, después de deducir ~USD 12.3 mil millones en incentivos a clientes que ofrece para fomentar el uso de su red. Sus ingresos provienen principalmente de tarifas cobradas a bancos y comercios por procesar transacciones en su red:

- Servicios (tarifas basadas en volumen de pago): ~USD 14.8 mil millones en el año fiscal 2023
- Procesamiento de datos (autorizaciones, compensación y liquidación): ~USD 16.0 mil millones
- Transacciones internacionales (comisiones por conversiones de moneda y pagos transfronterizos): ~USD 11.6 mil millones
- Otros ingresos (licencias, servicios de valor agregado como seguridad y datos): ~USD 2.5 mil millones

**Fuentes:**

- [Acerca de Visa](https://usa.visa.com/dam/VCOM/global/visa-everywhere/documents/visa-cashless-cities-report-esla.pdf#:~:text=para%20concretar%20pagos%20electrónicos%20rápidos%2C,Visa%20no%20es%20un%20banco)
- [Visa Inc. Reports Fiscal Fourth Quarter and Full-Year 2023 Results](https://s1.q4cdn.com/050606653/files/doc_financials/2023/q4/Q4-2023-Earnings-Release-FINAL.pdf#:~:text=Fiscal%20full,over)

## **The Goldman Sachs Group, Inc. (GS)**

**Actividad y sector:**

La Goldman Sachs Group, Inc. es uno de los principales bancos de inversión y proveedores de servicios financieros a nivel mundial. Su modelo de negocio se centra en tres segmentos principales: Global Banking & Markets (GBM), Asset & Wealth Management (AWM) y Platform Solutions. A través de estas divisiones, Goldman Sachs ofrece servicios de banca de inversión (asesoría en fusiones y adquisiciones, underwriting de emisiones de deuda y capital), intermediación en mercados globales (trading de instrumentos de renta fija, divisas, commodities y acciones), gestión de activos para inversionistas institucionales, gestión de patrimonios para clientes de alto patrimonio, así como financiamiento y préstamos privados. Más recientemente, mediante Platform Solutions, la firma ha incursionado en servicios financieros de consumo y tecnología financiera (por ejemplo, tarjetas de crédito en asociación y banca transaccional corporativa) buscando diversificar sus fuentes de ingreso más allá de sus operaciones tradicionales.

**Fuentes de Ingresos:**

- Global Banking & Markets (GBM): ~US$30,000 millones de ingresos netos en 2023. Es el mayor segmento de Goldman Sachs, que integra las actividades de banca de inversión y operaciones de mercados globales.

- Asset & Wealth Management (AWM): ~US$13,900 millones en 2023. Genera ingresos principalmente por comisiones de administración de activos y fondos de inversión, fees por desempeño e incentivo, servicios de gestión patrimonial y banca privada, así como por préstamos privados a clientes.

- Platform Solutions: ~US$2,400 millones en 2023. Corresponde a iniciativas de banca de consumo y tecnología financiera, incluyendo plataformas de tarjetas de crédito al consumo (ej., Apple Card) y negocios de banca transaccional para clientes corporativos. Esta unidad es relativamente nueva (creada tras una reorganización en 2022) y representa una pequeña porción de los ingresos totales, aunque de rápido crecimiento.

**Fuentes:**

- [How Goldman Sachs Makes Money](https://www.investopedia.com/articles/markets/041315/how-goldman-sachs-makes-its-money.asp#:~:text=What%20Exactly%20Does%20Goldman%20Sachs,Do)

## **The Progressive Corporation (PGR)**

**Actividad y sector:**

The Progressive Corporation es una compañía de seguros de Estados Unidos enfocada en el ramo de seguros de propiedad y accidentes (P&C). A través de sus subsidiarias de seguros, Progressive ofrece principalmente coberturas de automóviles personales y comerciales, seguros residenciales para viviendas, seguros de responsabilidad civil y propiedad comercial orientados a pequeñas empresas, seguros de compensación para trabajadores del sector transporte, entre otros productos de seguros especializados. La empresa opera en todo EE. UU. y distribuye sus pólizas tanto de forma directa (internet y teléfono) como mediante una amplia red de agentes independientes, adaptando sus programas a diversos mercados y canales de venta.

**Fuentes de Ingresos:**

Los ingresos de Progressive provienen casi totalmente de las primas de seguros (en 2023, el volumen total de primas netas emitidas de Progressive superó los $61,000 millones) que subscribe en tres segmentos operativos principales:

- Seguros de autos personales: representan ~79% de las primas netas emitidas por la compañía (aprox. $48,600 millones en 2023). Este segmento abarca pólizas para autos particulares y vehículos recreativos de clientes individuales.

- Seguros de líneas comerciales: aportan ~16% de las primas (cerca de $9,900 millones en 2023). Incluye seguros para vehículos comerciales, responsabilidad civil y propiedad para pequeñas empresas, así como camiones/flotas y compensación laboral principalmente en el sector transporte.

- Seguros de propiedad residencial: contribuyen ~5% de las primas (aprox. $3,100 millones en 2023). Comprende seguros de vivienda (hogar), inquilinos, casas móviles, así como participación en programas de seguro contra inundaciones bajo convenio federal.

**Fuentes:**

- [THE PROGRESSIVE CORPORATION 2023 ANNUAL REPORT](https://www.progressive.com/content/pdf/art/2023-annual-report.pdf#:~:text=in%20aggressive%20growth%20within%20our,7%20and%20rose)
- [Annual Report Pursuant to Section 13 or 15(d) of the Securities Exchange Act of 1934 For the fiscal year ended December 31, 2023](https://s202.q4cdn.com/605347829/files/doc_financials/2023/q4/14e21f85-2775-4649-87ed-35e8ca317f3e.pdf#:~:text=includes%20affiliates%20as%20well%29,casualty%20insurance%20and%20provide%20related)

In [2]:
tickers = ['BRK-B', 'V', 'GS', 'PGR']

prices = yf.download(tickers,start='2018-01-01', end='2025-06-03')['Close']

daily_rend = prices.pct_change().dropna()
rend = daily_rend.mean()

C:\Users\samue\AppData\Local\Temp\ipykernel_23904\2613940756.py:3: FutureWarning: YF.download() has changed argument auto_adjust default to True
  prices = yf.download(tickers,start='2018-01-01', end='2025-06-03')['Close']
[*********************100%***********************]  4 of 4 completed


In [3]:
# Matriz de pesos y Matriz de Varianza-Covarianza
w = np.array([0.25, 0.25, 0.25, 0.25])

cov = daily_rend.cov()

In [4]:
rend_p = float(np.dot(rend, w))
rend_diario = rend_p
rend_anual = rend_p * 252

df_rend = pd.DataFrame({
    "Métrica": ["Rendimiento Diario (%)", "Rendimiento Anual (%)"],
    "Valor": [rend_diario * 100, rend_anual * 100]
})

df_rend

Métrica      Valor
0  Rendimiento Diario (%)   0.081317
1   Rendimiento Anual (%)  20.491992

In [5]:
var_p = float(w.T @ cov @ w)
vol_diaria = np.sqrt(var_p)
vol_anual = np.sqrt(var_p * 252)

df_vol = pd.DataFrame({
    "Métrica": ["Volatilidad Diaria (%)", "Volatilidad Anual (%)"],
    "Valor": [vol_diaria * 100, vol_anual * 100]
})

df_vol

Métrica      Valor
0  Volatilidad Diaria (%)   1.347926
1   Volatilidad Anual (%)  21.397664

In [6]:
# Coeficiente de Variación
cv = float(vol_anual/rend_anual)
cv

1.0441963795115572

In [7]:
# Matriz de Correlación
cor = daily_rend.corr()
cor

Ticker     BRK-B        GS       PGR         V
Ticker                                        
BRK-B   1.000000  0.713607  0.534054  0.664746
GS      0.713607  1.000000  0.381716  0.590599
PGR     0.534054  0.381716  1.000000  0.424647
V       0.664746  0.590599  0.424647  1.000000

In [8]:
# Ratio de Sharpe
rs = float((rend_anual - 0.03) / vol_anual)
rs

0.8174720453999272

**Análisis de Portafolio:**

El análisis del portafolio muestra que el rendimiento esperado anual es positivo, lo que refleja la capacidad de la estrategia para generar valor con los activos seleccionados. En cuanto al riesgo, la volatilidad del portafolio resulta superior a la rentabilidad antes mencionada, indicando que la estrategia conlleva un nivel considerable pero no tan relevante de incertidumbre en los retornos, lo cuál se demuestra con un cv de ~1.04, lo que sugiere que por cada unidad de retorno esperado, se asume poco más unidad de riesgo. Asimismo, el Ratio de Sharpe calculado refleja la poca eficiencia de la estrategia de generar rendimientos sobre la libre de interés por unidad de riesgo asumido, pues con un RS de ~0.81, se encuentra por debajo del umbral de 1.0, lo que significa que el portafolio no compensa suficientemente el riesgo asumido con el exceso de retorno.

Finalmente, la matriz de correlación evidencia la presencia de correlaciones elevadas, lo que implica que los activos tienden a moverse en la misma dirección, reduciendo la posibilidad de que unos compensen las caídas de otros. En este sentido, el portafolio se encuentra expuesto a un mayor grado de riesgo no diversificable, ya que la similitud en los comportamientos de los activos limita la reducción del riesgo no sistemático total. Este comportamiento es consistente con la naturaleza de los instrumentos seleccionados, pues los cuatro pertenecen al sector financiero, haciendo que sus rendimientos respondan de manera similar a factores comunes de mercado, como tasas de interés, cambios regulatorios o ciclos económicos.

---

## 2. Comparación con estrategia pasiva
Ahora, compara tu estrategia de inversión activa contra una pasiva:

- En base a tu investigación preliminar busca un índice que represente el sector de tus activos, comenta el nombre, ticker y los constituyentes de este. Descarga los precios de Yahoo Finance. (10%)  
- Calcula las métricas de riesgo y rendimiento anuales, además del ratio de Sharpe. (20%)  

Nombre del Indice: MSCI US Investable Market Financials 25/50 Index

Vanguard Financials ETF, ticker: VFH

Principales constituyentes:

| Posición | Empresa               | Ticker | % del ETF |
| -------- | --------------------- | ------ | --------- |
| 1        | JPMorgan Chase & Co.  | JPM    | 9.75 %    |
| 2        | Berkshire Hathaway B  | BRK.B  | 7.44 %    |
| 3        | Mastercard Inc.       | MA     | 5.43 %    |
| 4        | Visa Inc.             | V      | 4.30 %    |
| 5        | Bank of America Corp. | BAC    | 4.02 %    |


Este índice está diseñado para reflejar el comportamiento del sector financiero de Estados Unidos, incluyendo empresas de gran, mediana y pequeña magnitud. Cubre bancos, aseguradoras, compañías de servicios financieros diversificados, proveedores de tarjetas de crédito y firmas de inversión.
El “25/50” en el nombre significa que ninguna empresa puede representar más del 25% del índice y que el conjunto de emisoras con más de 5% de peso no puede superar el 50% total. Esto evita que el índice esté demasiado concentrado en unas pocas empresas, asegurando una diversificación más balanceada dentro del sector.

El ETF VFH replica este índice, por lo que ofrece una exposición amplia al sector financiero de EUA con más de 400 compañías. Su objetivo es dar a los inversionistas una forma sencilla y de bajo costo de invertir en bancos, aseguradoras, procesadores de pagos y otros intermediarios financieros.
El fondo tiene la ventaja de ser líquido, diversificado y con gastos bajos, características típicas de los ETFs de Vanguard. Está muy ligado al ciclo económico: suele beneficiarse en entornos de crecimiento y alzas de tasas de interés, pero puede resentirse en periodos de crisis financieras o recesión.

- [MSCI US Investable Market Financials 25/50 Index - ETF Tracker](https://etfdb.com/index/msci-us-investable-market-financials-25-50-index/)

- [(VFH) Vanguard Financials ETF](https://www.vanguardmexico.com/es/producto/etf/renta-variable/0957/vanguard-financials-etf)

In [9]:
prices_VFH = yf.download('VFH',start='2018-01-01', end='2025-06-03')['Close']

rend_VFH = prices_VFH.pct_change().dropna()

C:\Users\samue\AppData\Local\Temp\ipykernel_23904\740864787.py:1: FutureWarning: YF.download() has changed argument auto_adjust default to True
  prices_VFH = yf.download('VFH',start='2018-01-01', end='2025-06-03')['Close']
[*********************100%***********************]  1 of 1 completed


In [10]:
# Rendimiento anual
rend_anual_VFH = rend_VFH.mean() * 252
rend_anual_VFH *100

Ticker
VFH    12.867205
dtype: float64

In [11]:
# Volatilidad anual
vol_anual_VFH = rend_VFH.std() * np.sqrt(252)
vol_anual_VFH * 100

Ticker
VFH    24.747652
dtype: float64

In [12]:
# Coeficiente de Variación
cv_VFH = float(vol_anual_VFH/rend_anual_VFH)
cv_VFH

C:\Users\samue\AppData\Local\Temp\ipykernel_23904\216495217.py:2: FutureWarning: Calling float on a single element Series is deprecated and will raise a TypeError in the future. Use float(ser.iloc[0]) instead
  cv_VFH = float(vol_anual_VFH/rend_anual_VFH)


1.9233122056258642

In [13]:
# Radio de Sharpe
rs_VFH = float((rend_anual_VFH - 0.03) / vol_anual_VFH)
rs_VFH

C:\Users\samue\AppData\Local\Temp\ipykernel_23904\2235118656.py:2: FutureWarning: Calling float on a single element Series is deprecated and will raise a TypeError in the future. Use float(ser.iloc[0]) instead
  rs_VFH = float((rend_anual_VFH - 0.03) / vol_anual_VFH)


0.39871276470184197

---

## 3. Conclusión
A manera de conclusión, en función de las métricas calculadas y en base a los dos Ratios de Sharpe calculados para cada estrategia, responde:

- ¿En qué estrategia invertirías y por qué?  
- ¿Habría una forma de mejorar la estrategia activa? Es decir, ¿qué podrías cambiar en tu portafolio para obtener mejores resultados? (30%)  

# Comparación de resultados

| Métrica                      | Estrategia Activa | Estrategia Pasiva |
| ---------------------------- | ------------------------------ | ------- |
| **Rendimiento anual (%)**    | 20.49                          | 12.86   |
| **Volatilidad anual (%)**    | 21.39                          | 24.74   |
| **Coeficiente de variación** | 1.04                           | 1.92    |
| **Ratio de Sharpe**          | 0.8174                         | 0.3987  |



Al comparar ambas estrategias, podemos observar que la estrategia activa supera a la pasiva en eficiencia. Esto se debe a que su rendimiento anual es mayor, su volatilidad es menor, y además tiene un coeficiente de variación más bajo, lo cual significa que aprovecha mejor el riesgo que toma. La métrica clave, el Ratio de Sharpe, también resulta más alto en el portafolio activo, confirmando que por cada unidad de riesgo se obtiene más rendimiento que con el ETF.

  La razón por la cuál nos basamos en los ratios de Sharpe es porque sirve para evaluar si el rendimiento de una inversión justifica el riesgo que se tomó para obtenerlo. Dicho de otra forma, no basta con fijarse en cuánto ganaste, sino si valió la pena el riesgo que corriste.

  Si tuvieramos que decidir, invertiríamos en el portafolio activo, porque sus resultados muestran que logra un mejor balance entre riesgo y beneficio. En cambio, el ETF, aunque es más diversificado y fácil de manejar, ofrece un rendimiento ajustado al riesgo más bajo.

  El tener un portafólio nos da la libertad de poder ajustar los pesos para mejorar nuestro rendimiento. La estrategia activa todavía puede mejorar. Una forma sería diversificar más los activos , incluyendo emisoras de otros sectores o regiones. También se podrían ajustar los pesos de las acciones en lugar de repartirlos por igual, favoreciendo a las que ofrecen mejor relación riesgo-rendimiento. Otra alternativa sería añadir activos de cobertura, como bonos o commodities, que ayuden a suavizar la volatilidad en momentos de crisis.

  En conclusión, aunque el ETF ofrece estabilidad y simplicidad, el portafolio activo destaca claramente en todas las métricas de desempeño ajustado. Con algunos ajustes, puede ser aún más sólido y competitivo como estrategia de inversión a largo plazo.

---

### Nota
Asume una tasa libre de riesgo del **3% anual**.

**Equipo:**
*   Samuel Gómez
*   Andrea Santoyo

